core/presentation/designsystem/src/main/java/com/pronoidsoftware/core/presentation/designsystem/components/TaskyDatePicker.kt

```diff
@@ -31,12 +30,11 @@ fun TaskyDatePicker(
     clock: Clock = Clock.System,
     content: @Composable () -> Unit,
 ) {
-    // don't use rememberDatePickerState because we want to be able to set the selected date
-    // from the date widget on the agenda overview screen too
-    val datePickerState = DatePickerState(
-        locale = Locale.getDefault(),
+    val datePickerState = rememberDatePickerState(
         initialSelectedDateMillis = selectedDate.toMillis(),
     )
+    // set to support cancelling
+    datePickerState.selectedDateMillis = selectedDate.toMillis()
```

Comment:
Now you have a side-effect again here :D In this case probably won't lead to any weird behavior, but the correct way would still be 
```
LaunchedEffect(selectedDate) {
    datePickerState.selectedDateMillis = selectedDate.toMillis()
}
```
so this really only gets executed when the selected date changes and not when anything changes.

Reply:
Ah, okay, thank you! I think I've refactored it now to avoid side effects